crates/transaction-fuzzer/src/account_universe/transfer_gen.rs

```diff
@@ -469,22 +469,24 @@ impl AUTransactionGen for P2PTransferGenRandomGasRandomPriceRandomSponsorship {
                     max_gas_price: PROTOCOL_CONFIG.max_gas_price(),
                 },
             }),
+            // NOTE: at protocol version 71, gas budget can be both too high and too low.
+            // So the variants here need to be ordered same as the checks in gas_v2.rs
```

Comment:
Just to make sure I understand, this changed because before we would have that `base_tx_cost_fixed * max_tx_gas_price < max_tx_gas` whereas this is no longer true with the new values for `max_tx_gas_price` and `max_tx_gas`. I just wanted to confirm that this is intended. 

Also, curious as to why/how these numbers were chosen. If there's not a strong reason, I'm wondering if lowering `max_tx_gas_price` to a lower value (e.g., to 10SUI) would be acceptable -- and this would have the benefit of maintaining this inequality. If you ask me why we need to maintain the inequality, I am not aware of anywhere that it is needed, but I would sleep more comfortably if we did maintain if it's no added work/wouldn't cause other issues.

Reply:
the invariant is interesting and we should go back and do some math. After all that is a non-sensical scenario. A transaction impossible to run?
We can also change values quickly as it does not appear to be an attack vector?

Reply:
Yes this is my understanding as well from reading the code [here](https://github.com/MystenLabs/sui/blob/2166353ad306493c5307f02b09ee63e99320c887/crates/sui-types/src/gas_model/gas_v2.rs#L313-L325). From [comments](https://github.com/MystenLabs/sui/blob/2166353ad306493c5307f02b09ee63e99320c887/crates/sui-types/src/gas_model/gas_v2.rs#L115-L121), it seems RGP is using `max_tx_gas_price` in some tests, so we saw cases where both `GasBudgetTooHigh` and `GasBudgetTooLow` errors can be triggered.
I think the `max_tx_gas` number was chosen to be as large as we are comfortable with, to allow as much bidding as possible. It seems `base_tx_cost_fixed` is `1_000`, so `max_tx_gas_price` can be 50 SUI instead? What do you think @dariorussi?

Reply:
Discussed with @dariorussi and we agreed to go with enforcing the invariant.